### PR TITLE
feat: added support for whitelisting custom CIDRs from SSM Parameters

### LINKED ISSUE #193: [FEATURE] Support PUBLIC_AND_PRIVATE and Public CIDR Range Filters for EKS Module
**Is your feature request related to a problem? Please describe.**
As an enterprise user, I would like to:
1. Have my Worker nodes communication stay within my VPC
2. Have the ability to filter the CIDR ranges that are allowed to connect to my EKS API Endpoint

**Describe the solution you'd like**
The current EKS module allows either private or public API endpoint. 
1. Add support for specifying PUBLIC_AND_PRIVATE as the target for the EndpointAccess Config
2. Add support for specifying multiple whitelists of CIDR ranges (stored in multiple SSM parameters) and potentially CIDRs from the public amazon ranges (https://docs.aws.amazon.com/vpc/latest/userguide/aws-ip-ranges.html). Concatenate these lists of CIDRs and add them to the EndpointAccess object via the `only_from` function (https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_eks/EndpointAccess.html#aws_cdk.aws_eks.EndpointAccess)

**Describe alternatives you've considered**
Creating a separate module that configures/locks down API network access. But this means the cluster will always be configured first with public access and then switched to limited access. 

**Additional context**
Customers running GuardDuty detect malicious access attempts from IP scanners that attempt to use guest/unauthenticated/anonymous credentials for accessing the EKS API. In order to limit the ability for scanners to be able to invoke the API, filtering based on whitelisted CIDRs is a simple and effective solution.

Switching purely to private endpoints has many knock on effects. For instance, it forced all seedfarmer modules that invoke kubectl to run within the VPC. It also requires all custom resources created by CDK to apply manifests to be run within the VPC. Finally it also requires either Cloud9 or a Bastion host to interact with the EKS API.

### PR DESCRIPTION
**Pull request by srinivasreddych**
Screenshot 2024-07-10 at 10 49 39 PM

*Issue #, if available:*
Fixes #193 

*Description of changes:*
- added support for whitelisting custom CIDRs from SSM Parameters and auto load AWS Codebuild IPs for being able to run seedfarmer commands. Attached a screenshot for reference.
- This improves the security posture of EKS API endpoint such that, nodes communicate privately and any communication to the EKS APIs will be allowed from whitelisted customer CIDRs and AWS CODEBUILD IPs (for being able to run seedfarmer commands outside of VPC)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 68289b65dc5cbab078bb9a3c774607aaddf480d7
* Result: FAILED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9b9549f5-6f98-4744-a090-940174b5e365%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 66c0751eb9f079068016b15e7b7a4c321b1a037b
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d0b77918-ff61-429a-a941-bb551c30c248%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/compute/eks/stack.py [hunk 1]
```diff
@@ -477,6 +482,16 @@ def _create_eks_cluster(
         Creates an Amazon EKS cluster with the specified configuration.
         """
 
+        if eks_compute_config.get("eks_api_endpoint_private"):
+            api_endpoint = eks.EndpointAccess.PRIVATE
+        else:
+            api_endpoint = eks.EndpointAccess.PUBLIC_AND_PRIVATE
```
Comment (dgraeber) [inline modules/compute/eks/stack.py:488]: What if the user is not passing in an SSM but the acutal CIDR's.  I think this locks the use case to only SSM parameters....

**Review by dgraeber — COMMENTED** [review]
See comments

File: modules/compute/eks/stack.py (hunk 1 above)
Comment (malachi-constant) [reply modules/compute/eks/stack.py:488]: Agreed, allow both

Comment (srinivasreddych) [reply modules/compute/eks/stack.py:488]: Good point.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: d1d894171be49c2d12dfd6642c9d3ba1663ca636
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=defcca1c-dc57-4680-ac2c-117287fd61b1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — DISMISSED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: f20d83b591c49eca3fa83337c79aa96d7994c68e
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ae7eacce-428d-4e11-988d-1c457a5f1315%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]

File: modules/compute/eks/stack.py [hunk 2]
```diff
@@ -477,6 +482,16 @@ def _create_eks_cluster(
...
+            if eks_compute_config.get("ips_to_whitelist"):
+                cidrs_from_ssm = self._fetch_cidrs_from_ssm(json.loads(eks_compute_config.get("ips_to_whitelist")))
```
Comment (dgraeber) [inline modules/compute/eks/stack.py]: I'm not sure I understand the logic.  If an SSM path is passed in get the cidrs, else NONE.  Also, fetch cidrs from the control plane. Merge them and apply.

What if customer wants to pass in a dedicated CIDR range in their VPC?  I may be misunderstanding here...so take with a grain of salt

**Review by dgraeber — COMMENTED** [review]
Left another comment....may be easier to hit me on slack

File: modules/compute/eks/stack.py (hunk 2 above)
Comment (srinivasreddych) [reply modules/compute/eks/stack.py]: As discussed and clarified offline, resolving the thread here

**Pull request merged**